odoo/addons/base/models/ir_actions.py

```diff
@@ -148,11 +148,17 @@ def _for_xml_id(self, full_xml_id):
         """
         record = self.env.ref(full_xml_id)
         assert isinstance(self.env[record._name], type(self))
-        action = record.sudo().read()[0]
+        return record._get_action_dict()
+
+    def _get_action_dict(self):
+        """ Returns the action content for the provided action record.
+        """
+        self.ensure_one()
+        readable_fields = self._get_readable_fields()
         return {
             field: value
-            for field, value in action.items()
-            if field in record._get_readable_fields()
+            for field, value in self.sudo().read()[0].items()
+            if field in readable_fields
```

Comment:
This doesn't seem to be used anywhere, either here or in the enterprise PR. What's the purpose, is it just to call `record._get_readable_fields()` only once? In that case, why not just extract that to a local and keep the rest untouched?

Reply:
It seems to be used here:
https://github.com/odoo-dev/enterprise/blob/718e74f4077e28b9546018e16a53147dc4d20d28/account_reports/models/account_report.py#L2940